src/vs/workbench/api/browser/mainThreadTreeViews.ts

```diff
@@ -219,19 +219,25 @@ export class TreeViewDataProvider implements ITreeViewDataProvider {
 		return this.itemsMap.size === 0;
 	}
 
-	private async postGetChildren(elements: ITreeItem[]): Promise<ResolvableTreeItem[]> {
+	// {{SQL CARBON EDIT}}
+	private async postGetChildren(elements: ITreeItem[]): Promise<ResolvableTreeItem[] | ITreeItem[]> {
 		const result: ResolvableTreeItem[] = [];
+		const results: ITreeItem[] = [];
 		const hasResolve = await this.hasResolve;
 		if (elements) {
 			for (const element of elements) {
-				const resolvable = new ResolvableTreeItem(element, hasResolve ? () => {
-					return this._proxy.$resolve(this.treeViewId, element.handle);
-				} : undefined);
 				this.itemsMap.set(element.handle, element);
-				result.push(resolvable);
+				if (hasResolve) {
```

Comment:
vscode changed the return type to ResolvableTreeItem just here and in all other places it still has ITreeItem. The dataProvider's itemsMap also still has the ITreeItem[] items but this method returns ResolvableTreeItem[] irrespective of hasResolve. When am trying to filter items, because of the the above, it returns 'Data tree node not found' error.

Reply:
Could you explain this further? I'm not understanding why this change is necessary and at first glance definitely seems like something we shouldn't be changing.

Reply:
Whenever a treeNode that is collapsed is selected, it goes through the following call stack: 
![image](https://user-images.githubusercontent.com/12754347/86656275-34365200-bf9c-11ea-8e59-258f02b5c672.png)

And the nodes in asyncDataTree are resolved from the above method postGetChildren which return ResolvableTreeItem but the dataProvider itemsMap has ITreeItem type. And on expand when it looks for the element to expand from dataProvider itemsMap it's not able to find the element since they're of two different types. 
The changes I made still returns ResolvableTreeItem if the dataProvider has resolveTreeItem defined. In my case it isn't defined and yet due to the above change, expand fails with 'Data tree node not found.'

Reply:
This works for the normal search view though, so what changed such that you need to do this for ours?